File: RiotGames/Dto/Match/PerkStyleSelectionDto.cs
using Newtonsoft.Json;

namespace potg.RiotGames.Dto.Match;

public sealed record PerkStyleSelectionDto
{
    [JsonProperty("perk")]
    public long Perk { get; init; }

    [JsonProperty("var1")]
    public long Var1 { get; init; }

    [JsonProperty("var2")]
    public long Var2 { get; init; }

    [JsonProperty("var3")]
    public long Var3 { get; init; }
}

File: src/potg/riot/lol/api/match/dto/perkStyleSelection.dto.ts
import { Field, ObjectType } from "@nestjs/graphql";

@ObjectType()
export class PerkStyleSelectionDto {
  @Field()
  perk: number;
  @Field()
  var1: number;
  @Field()
  var2: number;
  @Field()
  var3: number;
}
